openmc/mgxs/mgxs.py

```diff
@@ -4615,3 +4628,728 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
+    hdf5_key : str
+        The key used to index multi-group cross sections in an HDF5 data store
+
+    """
+
+    def __init__(self, domain=None, domain_type=None,
+                 groups=None, by_nuclide=False, name=''):
+        super(ChiPrompt, self).__init__(domain, domain_type, groups, by_nuclide, name)
+        self._rxn_type = 'chi-prompt'
+
+    @property
+    def scores(self):
+        return ['delayed-nu-fission', 'delayed-nu-fission',
+                'nu-fission', 'nu-fission']
+
+    @property
+    def filters(self):
+        # Create the non-domain specific Filters for the Tallies
+        group_edges = self.energy_groups.group_edges
+        energyout = openmc.Filter('energyout', group_edges)
+        energyin = openmc.Filter('energy', [group_edges[0], group_edges[-1]])
+        return [[energyin], [energyout], [energyin], [energyout]]
+
+    @property
+    def tally_keys(self):
+        return ['delayed-nu-fission-in', 'delayed-nu-fission-out',
+                'nu-fission-in', 'nu-fission-out']
+
+    @property
+    def rxn_rate_tally(self):
+        if self._rxn_rate_tally is None:
+            self._rxn_rate_tally = self.tallies['nu-fission-out'] - \
+                                   self.tallies['delayed-nu-fission-out']
+            self._rxn_rate_tally.sparse = self.sparse
+        return self._rxn_rate_tally
+
+    @property
+    def xs_tally(self):
+
+        if self._xs_tally is None:
+            delayed_nu_fission_in = self.tallies['delayed-nu-fission-in']
+            nu_fission_in = self.tallies['nu-fission-in']
+            prompt_nu_fission_in = nu_fission_in - delayed_nu_fission_in
+
+            # Remove coarse energy filter to keep it out of tally arithmetic
+            energy_filter = prompt_nu_fission_in.find_filter('energy')
+            prompt_nu_fission_in.remove_filter(energy_filter)
+
+            # Compute chi
+            self._xs_tally = self.rxn_rate_tally / prompt_nu_fission_in
+            super(ChiPrompt, self)._compute_xs()
+
+            # Add the coarse energy filter back to the nu-fission tally
+            prompt_nu_fission_in.filters.append(energy_filter)
+
+        return self._xs_tally
+
+    def get_slice(self, nuclides=[], groups=[]):
+        """Build a sliced ChiPrompt for the specified nuclides and energy
+           groups.
+
+        This method constructs a new MGXS to encapsulate a subset of the data
+        represented by this MGXS. The subset of data to include in the tally
+        slice is determined by the nuclides and energy groups specified in
+        the input parameters.
+
+        Parameters
+        ----------
+        nuclides : list of str
+            A list of nuclide name strings
+            (e.g., ['U-235', 'U-238']; default is [])
+        groups : list of Integral
+            A list of energy group indices starting at 1 for the high energies
+            (e.g., [1, 2, 3]; default is [])
+
+        Returns
+        -------
+        openmc.mgxs.MGXS
```

Comment:
Make the return value `openmc.mgxs.ChiPrompt`. I think it is more generally `openmc.mgxs.MGXS` for the general `get_slice(...)` method implemented at the abstract `MGXS` class level.